pkg/daemon/ceph/osd/volume.go

```diff
@@ -586,12 +579,6 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 
 			logger.Infof("configuring new LVM device %s", name)
 			deviceArg := path.Join("/dev", name)
-			// ceph-volume prefers to use /dev/mapper/<name> if the device has this kind of alias
-			for _, devlink := range device.PersistentDevicePaths {
-				if strings.HasPrefix(devlink, "/dev/mapper") {
```

Comment:
> @travisn
> 
> > Shouldn't we remove the dev mappings handling code for OSDs on PVCs? Or why would we need it?
> 
> You're right. This logic is not necessary. So this patch makes sense not as a bug fix but as a cleanup. How about merging this PR anyway? In addition, I'll submit a PR to handle `osdsPerDevice` properly.

I wonder if there is still a scenario where we need to keep the /dev/mapper detection. If the user has specified useAllDevices or similar settings, the /dev/mapper path is still preferred. But in the case where the user has specified explicit paths such as with local PVs or named devices in the cluster CR, there is no need for this search since they can be specified already as such with the /dev/mapper path. If we add this check back in for the non-pvc case here, we should add it for raw mode in the `initializeDevicesRawMode()` method where it is missing now. Or perhaps we keep it simple, consistent, and predictable by merging this PR without further changes.
@satoru-takeuchi @ausias-armesto thoughts?

Reply:
@travisn I only focused on LV-backed PVC by mistake and now I have a concern.

This logic is not necessary for LV-backed OSD for both OSD on device 
nd OSD on PVC because this configuration always uses raw mode and `ceph-volume raw prepare` accepts all device names. However, if a new OSD creation uses lvm mode, /dev/mapper prefix must be used because `ceph-volume lvm prepare` only accept the device names prefixed by /dev/mapper. This limitation exists for both OSD on device and OSD on PCV.

Please wait for a while. I'll verify my concern later.

Reply:
@travisn , I can only test on LV-backed PVC, and cannot provide further information about how would behave in other contexts. On the other side, changing `osdsPerDevice ` to `1` has not changed anything, is still not working.

Thanks for following this issue

Reply:
@ausias-armesto Could you show me your cluster.yaml and the log of osd prepare pod when you specified `osdsPerDevice: 1`?

Reply:
[osd-prepare-1.log](https://github.com/rook/rook/files/10192870/osd-prepare-1.log)
[osd-prepare-2.log](https://github.com/rook/rook/files/10192871/osd-prepare-2.log)
[osd-prepare-3.log](https://github.com/rook/rook/files/10192873/osd-prepare-3.log)
[helm-cluster.yaml.txt](https://github.com/rook/rook/files/10192942/helm-cluster.yaml.txt)

The disk partitions for servers running on osd2 and osd3 looks like this:
```
root@server-stage-fsn1-002 /home/debian # lsblk -f
NAME           FSTYPE            FSVER    LABEL    UUID                                   FSAVAIL FSUSE% MOUNTPOINT
sda                                                                                                      
├─sda1         linux_raid_member 1.2      rescue:0 492095b8-f9ee-e7c0-9e1e-e19193d3c00f                  
│ └─md0        ext3              1.0               ce3f530e-9eb8-4d67-ac74-44fe208f9328    877.3M     6% /boot
└─sda2         linux_raid_member 1.2      rescue:1 d9d405cd-528f-6f96-99b3-9b7fb7469523                  
  └─md1        LVM2_member       LVM2 001          ECx2sJ-9et7-Sg4F-oC59-nge2-LLDQ-3WACPR                
    ├─vg0-root ext4              1.0               0545917f-d52f-4902-81f6-88c5f02e84ba     37.9G    17% /
    ├─vg0-swap swap              1                 236c4901-9f82-4399-a8d6-b61147818169                  
    ├─vg0-tmp  reiserfs          3.6               9b4771e7-73a5-41f9-ab4f-56972f9be3e1      4.6G     8% /tmp
    └─vg0-data                                                                                           
sdb                                                                                                      
├─sdb1         linux_raid_member 1.2      rescue:0 492095b8-f9ee-e7c0-9e1e-e19193d3c00f                  
│ └─md0        ext3              1.0               ce3f530e-9eb8-4d67-ac74-44fe208f9328    877.3M     6% /boot
└─sdb2         linux_raid_member 1.2      rescue:1 d9d405cd-528f-6f96-99b3-9b7fb7469523                  
  └─md1        LVM2_member       LVM2 001          ECx2sJ-9et7-Sg4F-oC59-nge2-LLDQ-3WACPR                
    ├─vg0-root ext4              1.0               0545917f-d52f-4902-81f6-88c5f02e84ba     37.9G    17% /
    ├─vg0-swap swap              1                 236c4901-9f82-4399-a8d6-b61147818169                  
    ├─vg0-tmp  reiserfs          3.6               9b4771e7-73a5-41f9-ab4f-56972f9be3e1      4.6G     8% /tmp
    └─vg0-data                                                                                           
```


The server running on osd1, has 2 extra disks `sdc` and `sdd` which are correctly detected.